File: backend/src/DTO/BlocDto.php
<?php

namespace App\DTO;

use Symfony\Component\Validator\Constraints as Assert;

class BlocDto
{
    public function __construct(
      #[Assert\NotBlank]
      public string $id,

      #[Assert\NotBlank]
      public string $name,

      public ?string $description,

      public ?string $blocLowRes,

      public ?string $blocMedRes,

      public ?string $blocHighRes)
    { }
}

File: backend_net/Model/Dto/BlocDto.cs
using System;

namespace Thecell.Bibaboulder.Model.Dto;

public class BlocDto
{
    public required Guid Id { get; set; }
    public required string Name { get; set; }
    public string? Description { get; set; }
    public string? BlocLowRes { get; set; }
    public string? BlocMedRes { get; set; }
    public string? BlocHighRes { get; set; }
}
